public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,879 @@
+  Now create the component's template, `app/hero-detail.component.html`, with the following markup.
+
++makeExample('reactive-forms/ts/app/hero-detail-1.component.html', 'basic-form','app/hero-detail.component.html')(format=".")
+
+:marked
+  The `novalidate` attribute in the `<form>` element prevents the browser 
+  from attempting native HTML validations. 
+  The `[formGroup]="heroForm"` tells Angular that the form element
+  is associated with the `heroForm` property of the component class.
+  The `formControlName="name"` associates the `input` element with the `name` control inside the `heroForm`.
+
+.l-sub-section
+  :marked
+    Disregard the `form-group` and `form-control` _CSS_ classes. They belong to the 
+    <a href="http://getbootstrap.com/" target="_blank" title="Bootstrap CSS">Bootstrap CSS library</a>,
+    not Angular.
+    They _style_ the form but in no way impact the logic of the form.
+
+a#import
+:marked
+  ## Import the _ReactiveFormsModule_
+
+  The `HeroDetailComponent` template uses directives from 
+  the `ReactiveFormsModule` in the `@angular/forms` library
+  so you must add it to the `imports` array of the `NgModule` that declares that component.
+  
+  In this sample, you declare the `HeroDetailComponent` in the `AppModule`.
+  Therefore, do the following three things in `app.module.ts`:
+  1. Use a JavaScript `import` statement to access 
+  the `ReactiveFormsModule` and the `HeroDetailComponent`.
+  1. Add `ReactiveFormsModule` to the `AppModule`'s `imports` list.
+  1. Add `HeroDetailComponent` to the declarations array.
+
++makeExample('reactive-forms/ts/app/app.module.ts', 'v1','app/app.module.ts (excerpt)')(format=".")
+
+a#update
+.l-main-section
+:marked
+  ## Display the _HeroDetailComponent_
+  Revise the `AppComponent` template so it displays the `HeroDetailComponent`.
++makeExample('reactive-forms/ts/app/app.component.1.ts', '','app/app.component.ts')(format=".")
+
+:marked
+  ### Style the app
+  You used bootstrap CSS classes in the template HTML of both the `AppComponent` and the `HeroDetailComponent`.
+  Add the `bootstrap` _CSS stylesheet_ to the head of `index.html`:
+
++makeExample('reactive-forms/ts/index.html', 'bootstrap','index.html')(format=".")
+
+:marked
+  Now that everything is wired up, the browser should display something like this: 
+
+figure.image-display
+  img(src="/resources/images/devguide/reactive-forms/hero-detail.png" width="400px" alt="Basic reactive form")
+
+:marked
+  Great! You have the basics of a form. 
+  
+  In real life apps, forms get big fast. 
+  The way you are creating `FormControls` currently is both tedious and hard to maintain. 
+  `FormBuilder` makes form development and maintenance easier.
+
+
+.l-main-section
+a#formbuilder
+:marked
+  ## Introduction to _FormBuilder_
+
+  The `FormBuilder` class helps reduce repetition and 
+  clutter by handling details of control creation for you. 
+
+  You already imported `FormBuilder` into `hero-detail.component.ts`.
```

Comment:
We could instead have them import FormBuilder here. Then the reader is not importing so much that is not needed at the beginning.